health-monitors/kubernetes-object-monitor/pkg/initializer/initializer.go

```diff
@@ -0,0 +1,232 @@
+	Config    *config.Config
+}
+
+func InitializeAll(ctx context.Context, params Params) (*Components, error) {
+	cfg, err := config.Load(params.PolicyConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load policy config: %w", err)
+	}
+
+	slog.Info("Loaded policy configuration", "policies", len(cfg.Policies))
+
+	conn, err := dialPlatformConnector(params.PlatformConnectorSocket)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to platform connector: %w", err)
+	}
+
+	pcClient := pb.NewPlatformConnectorClient(conn)
+	pub := publisher.New(pcClient)
+
+	mgr, err := createManager(params)
+	if err != nil {
+		conn.Close()
+		return nil, fmt.Errorf("failed to create manager: %w", err)
+	}
+
+	if err := setupHealthChecks(mgr); err != nil {
+		conn.Close()
+		return nil, fmt.Errorf("failed to setup health checks: %w", err)
+	}
+
+	celEnv, err := celenv.NewEnvironment(mgr.GetClient())
+	if err != nil {
+		conn.Close()
+		return nil, fmt.Errorf("failed to create CEL environment: %w", err)
+	}
+
+	evaluator, err := policy.NewEvaluator(celEnv, cfg.Policies)
+	if err != nil {
+		conn.Close()
+		return nil, fmt.Errorf("failed to create policy evaluator: %w", err)
+	}
+
+	if err := registerControllers(mgr, evaluator, pub, cfg.Policies, params.MaxConcurrentReconciles); err != nil {
+		conn.Close()
+		return nil, fmt.Errorf("failed to register controllers: %w", err)
+	}
+
+	return &Components{
+		Manager:   mgr,
+		GRPCConn:  conn,
+		Publisher: pub,
+		Evaluator: evaluator,
+		Config:    cfg,
+	}, nil
+}
+
+func createManager(params Params) (ctrl.Manager, error) {
+	config := ctrl.GetConfigOrDie()
+
+	mgrOpts := ctrl.Options{
+		Metrics: server.Options{
+			BindAddress: params.MetricsBindAddress,
+		},
+		HealthProbeBindAddress: params.HealthProbeBindAddress,
+		Cache: cache.Options{
+			SyncPeriod: &params.ResyncPeriod,
+		},
+	}
+
+	mgr, err := ctrl.NewManager(config, mgrOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create manager: %w", err)
+	}
+
+	return mgr, nil
+}
+
+func setupHealthChecks(mgr ctrl.Manager) error {
+	if err := mgr.AddHealthzCheck("ping", func(req *http.Request) error { return nil }); err != nil {
+		return fmt.Errorf("failed to add health check: %w", err)
+	}
+
+	if err := mgr.AddReadyzCheck("ping", func(req *http.Request) error { return nil }); err != nil {
+		return fmt.Errorf("failed to add ready check: %w", err)
+	}
+
+	return nil
+}
+
+func registerControllers(
+	mgr ctrl.Manager,
+	evaluator *policy.Evaluator,
+	pub *publisher.Publisher,
+	policies []config.Policy,
+	maxConcurrentReconciles int,
+) error {
+	gvkPolicies := groupPoliciesByGVK(policies)
+
+	for gvk, policies := range gvkPolicies {
+		reconciler := controller.NewResourceReconciler(mgr.GetClient(), evaluator, pub, policies, gvk)
+
+		if err := ctrl.NewControllerManagedBy(mgr).
+			For(newUnstructuredForGVK(gvk)).
+			WithOptions(ctrlcontroller.Options{
+				MaxConcurrentReconciles: maxConcurrentReconciles,
+			}).
```

Comment:
Correct me if I'm wrong, but in the design, I believe we said the `maxConcurrentReconciles` would be for per-resource parallelism. In this case it seems to be more like per object parallelism.  Since we start all the controller managers in a loop, I believe all the gkvs will be reconciled in parallel -- which is fine tbh. I don't really mind having all the resources reconcile in parallel since we shouldn't have very many of them.